lib/oban/worker.ex

```diff
@@ -138,6 +138,13 @@ defmodule Oban.Worker do
   """
   @callback backoff(attempt :: pos_integer()) :: pos_integer()
 
+  @doc """
+  Maximimum job execution time limit
+
+  Defaults to `:infinity`
+  """
+  @callback timeout(job :: Job.t()) :: pos_integer() | :infinity
```

Comment:
The built in `timeout` type uses `non_neg_integer()` instead of `pos_integer()`. I doubt there is any practical difference between 0ms and 1ms, but I think the type should match.

Using the actual `timeout` type seems a little odd for a function named `timeout`.